python/felis/datamodel.py

```diff
@@ -225,6 +225,31 @@ class Column(BaseObject):
     votable_datatype: str | None = Field(None, alias="votable:datatype")
     """The VOTable datatype of the column."""
 
+    @model_validator(mode="after")
+    def check_value(self) -> Column:
+        """Check that the default value is valid."""
+        if (value := self.value) is not None:
+            if value is not None and self.autoincrement is True:
+                raise ValueError("Column cannot have both a default value and be autoincremented")
+            felis_type = FelisType.felis_type(self.datatype)
+            if felis_type.is_numeric:
+                if felis_type in (Byte, Short, Int, Long):
+                    if not isinstance(value, int):
+                        raise ValueError("Default value must be an int for integer type columns")
+                elif felis_type in (Float, Double):
+                    if not isinstance(value, float):
+                        raise ValueError(
+                            "Default value must be a decimal number for float and double columns"
+                        )
```

Comment:
Again, maybe add `else` with exception, to be sure that you handle all type in the above two clauses.

Reply:
I don't think an `else` is necessary here because all of the Felis numeric types are explicitly handled. It should not be possible for `is_numeric` to be `True` and `felis_type` to not be one of these six types.